ledger-service/http-json/src/it/scala/http/WebsocketServiceIntegrationTest.scala

```diff
@@ -450,6 +450,83 @@ class WebsocketServiceIntegrationTest
       } yield resumes.foldLeft(1 shouldBe 1)((_, a) => a)
   }
 
+  "fetch multiple keys should work" in withHttpService { (uri, encoder, _) =>
+    def matches(expected: Seq[JsValue], actual: Seq[JsValue]): Boolean =
+      expected.length == actual.length && (expected, actual).zipped.forall {
+        case (exp, act) => matchesJs(exp, act)
+      }
+    // matches if all the values specified in expected appear with the same
+    // value in actual; actual is allowed to have extra fields. Arrays must
+    // have the same length.
+    def matchesJs(expected: spray.json.JsValue, actual: spray.json.JsValue): Boolean = {
+      import spray.json._
+      (expected, actual) match {
+        case (JsArray(expected), JsArray(actual)) =>
+          expected.length == actual.length && matches(expected, actual)
+        case (JsObject(expected), JsObject(actual)) =>
+          expected.keys.forall(k => matchesJs(expected(k), actual(k)))
+        case (JsString(expected), JsString(actual)) => expected == actual
+        case (JsNumber(expected), JsNumber(actual)) => expected == actual
+        case (JsBoolean(expected), JsBoolean(actual)) => expected == actual
+        case (JsNull, JsNull) => true
+        case _ => false
+      }
+    }
+    def create(account: String): Future[domain.ContractId] =
+      for {
+        r <- postCreateCommand(accountCreateCommand(domain.Party("Alice"), account), encoder, uri)
+      } yield {
+        assert(r._1.isSuccess)
+        getContractId(getResult(r._2))
+      }
+    def archive(id: domain.ContractId): Future[Assertion] =
+      for {
+        r <- postArchiveCommand(domain.TemplateId(None, "Account", "Account"), id, encoder, uri)
+      } yield {
+        assert(r._1.isSuccess)
+      }
+    val req =
+      """
+          |[{"templateId": "Account:Account", "key": ["Alice", "abc123"]},
+          | {"templateId": "Account:Account", "key": ["Alice", "def456"]}]
+          |""".stripMargin
+    val futureResults =
+      singleClientFetchStream(jwt, uri, req).via(parseResp).runWith(Sink.seq[JsValue])
+
+    for {
+      cid1 <- create("abc123")
+      _ <- create("abc124")
+      _ <- create("abc125")
+      cid2 <- create("def456")
+      _ <- archive(cid2)
+      _ <- archive(cid1)
+      results <- futureResults
+    } yield {
+      val expected: Seq[JsValue] = {
+        import spray.json._
+        Seq(
+          """
+            |{"events": []}
```

Comment:
@garyverhaegen-da are you sure this test works... I see a space before `[]` which should NOT be there... In any case I don't like tests relying on formatted JSON string.